command/command.ts

```diff
@@ -1601,7 +1601,12 @@ export class Command<
             arg = parseArgValue(args.shift() as string);
           }
 
-          if (typeof arg !== "undefined") {
+          if (
+            expectedArg.variadic && this.getArguments().length === 1 &&
```

Comment:
If arguments was defined with one or more positional arguments, with a variadic argument at the end (`<arg1> <arg2> <others...>`, I figured the user would want to access the elements of the variadic portion as a list (`["foo", "bar", ["one", "two", "three"] ]`) instead of having them appended to one flat array.

As an example:

```typescript
const { args } = await new Command()
  .name("copy file")
  .arguments("<fileToCopy> <...destinations>")
  .parse(Deno.args);

console.log(args)
```

```sh
$ deno run ./cp.ts ./notes.txt ~/Documents/notes.txt ~/Desktop/notes.txt
> [
    "./notes.txt",
    [
      "/Users/averypierce/Documents/notes.txt",
      "/Users/averypierce/Desktop/notes.txt"
    ]
  ]
```

In this case, it's helpful for `destinations` to already be rolled-up into an array, because I'm going to perform the same operation with each one.

Reply:
Here's another example which I hope illustrates the benefit of keeping variadic args rolled-up:

```typescript
const { args } = await new Command()
  .name("Sendmail")
  .arguments("<subject> <body> <...recipients>")
  .parse(Deno.args);

console.log(args)
```

```
$ deno run sendmail.ts "Hello World" "Thank you for subscribing to my mailing list" alice@example.com bob@example.com charlie@example.com
> [
    "Hello World",
    "Thank you for subscribing to my mailing list",
    [ "alice@example.com", "bob@example.com", "charlie@example.com" ]
  ]
```

Reply:
That was originally the reason I implemented it the way it is now. But i think it can be confusing if we have two different behaviours for rest arguments depending on how many arguments are defined. And i think it's nicer if the syntax of the action handler arguments and the arguments defined with the arguments method are the same. It's easier to read.

```ts
const { args: [arg1, arg2, ...restArgs] } = await new Command()
  .arguments("[arg1] [arg2] [...restArgs]")
  .action((options, arg1, arg2, ...restArgs) => {})
  .parse();
```